### PR TITLE
Make emissions tracker context manager

### LINKED ISSUE #137: Add context manager
Would be nice to be able to do something like 

```python
with EmissionsTracker() as tracker:
	...
```

### PR DESCRIPTION
**Pull request by connor-mccarthy**
closes #137

@benoit-cty Could you please review and provide your feedback?

The [code changes](https://github.com/mlco2/codecarbon/blob/ab24732a9a91967e1a0c60161a3a8faffe87ef83/codecarbon/emissions_tracker.py#L53) are fairly minimal and the [example](https://github.com/mlco2/codecarbon/blob/ab24732a9a91967e1a0c60161a3a8faffe87ef83/examples/mnist_context_manager.py) should demonstrate the fairly straight-forward implementation.

I believe the only somewhat opinionated change is this line [here](https://github.com/mlco2/codecarbon/blob/ab24732a9a91967e1a0c60161a3a8faffe87ef83/codecarbon/emissions_tracker.py#L281). Perhaps you'd want to name that property something different. I deliberately did not choose `BaseEmissionsTracker.emissions` to avoid implying a getter/setter pattern with the existing `BaseEmissionsTracker._emissions` property.

### CODE REVIEW

**Review by vict0rsch — APPROVED** [review]

**Comment by vict0rsch** [comment]
@Lothiraldan any idea why the conda builds are failing?

**Comment by vict0rsch** [comment]
Thanks @connor-mccarthy ! LGTM
Let's just fix those tests before we merge

**Comment by benoit-cty** [comment]
Thanks you very much for your contribution.

**Comment by benoit-cty** [comment]
@vict0rsch "HTTP 502 BAD GATEWAY for url" mean the reverse proxy don't find the service.
 It's probably a temporary error.

**Comment by connor-mccarthy** [comment]
@benoit-cty @vict0rsch I believe I've resolved the failing tests for Python 3.6.

File: README.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-![banner](docs/edit/images/banner.png)
+****![banner](docs/edit/images/banner.png)
```
Comment (benoit-cty) [inline README.md:1]: Why did you add "****" ?

**Review by benoit-cty — COMMENTED** [review]
Beside the "****" that I don't understand it's ok for.
I've tested your code and it run as expected.

File: README.md (hunk 1 above)
Comment (connor-mccarthy) [reply README.md]: Resolved! I'm sure I just hit command+b by accident.

**Review by benoit-cty — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (benoit-cty) [reply README.md]: You're welcome.

**Comment by benoit-cty** [comment]
Thank you very much for your contribution.
